tests/rptest/tests/limits_test.py

```diff
@@ -22,34 +22,89 @@
 from rptest.util import wait_until
 
 
+def _timed_out(e: RpkException):
+    return "timed out" in e.msg
+
+
+def _message_to_large(e: RpkException):
+    return "MESSAGE_TOO_LARGE" in e.stderr
+
+
 class LimitsTest(RedpandaTest):
     def _rand_string(sz):
         return ''.join(
             random.choice(string.ascii_letters + string.digits)
             for _ in range(sz))
 
+    def assert_produce_result(self, topic, size, failure_predicate=None):
+        rpk = RpkTool(self.redpanda)
+
+        def expected_result():
+            try:
+                rpk.produce(topic=topic,
+                            key=LimitsTest._rand_string(64),
+                            msg=LimitsTest._rand_string(size))
+            except RpkException as e:
+                return failure_predicate and failure_predicate(e)
+
+            return failure_predicate is None
+
+        wait_until(expected_result, 30, 1)
+
+    def change_config(self, key, value):
```

Comment:
nit: I don't think this helper adds readability. It is used only once and obfuscates a bit if cluster config or topic config is changed.